packages/rlc-common/src/buildObjectTypes.ts

```diff
@@ -419,16 +423,29 @@ export function getPropertySignature(
   const propertyName = property.name;
 
   const description = property.description;
-  let type =
-    generateForOutput(schemaUsage, property.usage) && property.outputTypeName
-      ? property.outputTypeName
-      : property.typeName
-      ? property.typeName
-      : property.type;
-  if (property.typeName && property.fromCore) {
-    importedModels.add(property.typeName);
+  let type;
+  if (
+    property.type === "array" &&
+    (property as any).items.fromCore &&
+    property.typeName
+  ) {
```

Comment:
Consider the case of Record<Error>  ?

Reply:
Yes, record is a valid case we could add it.

Reply:
Added in UT.